torch/lib/c10d/reducer.cpp

```diff
@@ -1469,6 +1515,90 @@ void Reducer::ensure_prior_reduction_finished() {
   }
 }
 
+bool Reducer::should_collect_runtime_stats() {
+  if (num_iterations_ > 0 && num_iterations_ % kDDPRuntimeLoggingSampleRate == 0) {
```

Comment:
this means that the first time we will collect runtime stats is iteration 100? Should we make an exception and collect them at iteration 1 as well, so users can view some stats before iteration 100 (maybe for debugging purposes)?

Reply:
sounds good